### PR TITLE
Custom check to check for RavenDB index errors and reports errors as log entry and custom check failure

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

**Review by tmasternak — APPROVED** [review]

**Comment by danielmarbach** [comment]
@ramonsmits and I had a chat and agreed we will be turning this check into a custom check for now to make it less invasive for now until we get a better understanding of when indexes go into the error state

**Review by mikeminutillo — APPROVED** [review]

**Review by tmasternak — APPROVED** [review]

File: src/ServiceControl.Audit/Infrastructure/CheckRavenDBIndexErrors.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            _store = store;
+        }
+
+        public override Task<CheckResult> PerformCheck()
+        {
+            var statistics = _store.DatabaseCommands.GetStatistics();
+            var indexes = statistics.Indexes.OrderBy(x => x.Name).ToArray();
+
+            if (statistics.Errors.Length <= 0)
```
Comment (SzymonPobiega) [inline src/ServiceControl.Audit/Infrastructure/CheckRavenDBIndexErrors.cs:24]:
```suggestion
            if (statistics.Errors.Length = 0)
```

Nitpicking but it feels odd if length is negative

File: src/ServiceControl/Operations/CheckRavenDBIndexErrors.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            _store = store;
+        }
+
+        public override Task<CheckResult> PerformCheck()
+        {
+            var statistics = _store.DatabaseCommands.GetStatistics();
+            var indexes = statistics.Indexes.OrderBy(x => x.Name).ToArray();
+
+            if (statistics.Errors.Length <= 0)
```
Comment (SzymonPobiega) [inline src/ServiceControl/Operations/CheckRavenDBIndexErrors.cs:24]:
```suggestion
            if (statistics.Errors.Length = 0)
```

Nitpicking

**Review by SzymonPobiega — APPROVED** [review]
Two minor comments :)

File: src/ServiceControl/Operations/CheckRavenDBIndexErrors.cs (hunk 2 above)
Comment (danielmarbach) [reply src/ServiceControl/Operations/CheckRavenDBIndexErrors.cs:24]: But then the code should at least compile ;) We are not writing javascript here ;)

**Review by danielmarbach — APPROVED** [review]

**Pull request merged**